static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.jsx

```diff
@@ -1,71 +1,101 @@
+    if (!authToken) return
 
-export class DownloadHistoryContainer extends Component {
-  componentDidMount() {
-    const {
-      onFetchRetrievalHistory
-    } = this.props
+    setRetrievalHistoryLoading(true)
 
-    onFetchRetrievalHistory()
+    try {
+      const requestObject = new RetrievalRequest(authToken, earthdataEnvironment)
+      const response = await requestObject.all()
+      const { data } = response
+
+      setRetrievalHistory(data)
+      setRetrievalHistoryLoaded(true)
+    } catch (error) {
+      dispatchHandleError({
+        error,
+        action: 'fetchRetrievalHistory',
+        resource: 'retrieval history',
+        verb: 'fetching',
+        notificationType: 'banner'
+      })
+    } finally {
+      setRetrievalHistoryLoading(false)
+    }
   }
 
-  render() {
-    const {
-      earthdataEnvironment,
-      retrievalHistory,
-      onDeleteRetrieval,
-      retrievalHistoryLoading,
-      retrievalHistoryLoaded
-    } = this.props
-
-    return (
-      <DownloadHistory
-        earthdataEnvironment={earthdataEnvironment}
-        retrievalHistory={retrievalHistory}
-        retrievalHistoryLoading={retrievalHistoryLoading}
-        retrievalHistoryLoaded={retrievalHistoryLoaded}
-        onDeleteRetrieval={onDeleteRetrieval}
-      />
-    )
+  const handleDeleteRetrieval = async (retrievalId) => {
+    if (!authToken) return
+
+    try {
+      const requestObject = new RetrievalRequest(authToken, earthdataEnvironment)
+      await requestObject.remove(retrievalId)
+
+      setRetrievalHistory((prevHistory) => prevHistory.filter((item) => item.id !== retrievalId))
+
+      addToast('Retrieval removed', {
+        appearance: 'success',
+        autoDismiss: true
+      })
+    } catch (error) {
+      dispatchHandleError({
+        error,
+        action: 'handleDeleteRetrieval',
+        resource: 'retrieval',
+        verb: 'deleting',
+        notificationType: 'banner'
+      })
+    }
   }
-}
 
-DownloadHistoryContainer.defaultProps = {
-  retrievalHistory: []
+  useEffect(() => {
+    fetchRetrievalHistory()
+  }, [earthdataEnvironment, authToken])
+
+  return (
+    <DownloadHistory
+      earthdataEnvironment={earthdataEnvironment}
+      retrievalHistory={retrievalHistory}
+      retrievalHistoryLoading={retrievalHistoryLoading}
+      retrievalHistoryLoaded={retrievalHistoryLoaded}
+      onDeleteRetrieval={handleDeleteRetrieval}
+    />
+  )
 }
 
 DownloadHistoryContainer.propTypes = {
+  authToken: PropTypes.string.isRequired,
   earthdataEnvironment: PropTypes.string.isRequired,
-  retrievalHistory: PropTypes.arrayOf(
-    PropTypes.shape({})
-  ),
-  retrievalHistoryLoading: PropTypes.bool.isRequired,
-  retrievalHistoryLoaded: PropTypes.bool.isRequired,
-  onFetchRetrievalHistory: PropTypes.func.isRequired,
-  onDeleteRetrieval: PropTypes.func.isRequired
+  dispatchHandleError: PropTypes.func.isRequired
 }
 
+const mapStateToProps = (state) => ({
```

Comment:
Every other container puts `mapStateToProps` and `mapDispatchToProps` before the component, please follow existing patterns in the code